src/test/java/io/antmedia/test/StreamFetcherUnitTest.java

```diff
@@ -216,6 +225,85 @@ public void testBugUpdateStreamFetcherStatus() {
 		logger.info("leaving testBugUpdateStreamFetcherStatus");
 
 	}
+	
+	
+	@Test
+	public void testRestartPeriodStreamFetcher() {
+
+		try {
+			//Create Stream Fetcher Manager
+			QuartzSchedulingService scheduler = (QuartzSchedulingService) applicationContext.getBean(QuartzSchedulingService.BEAN_NAME);
+
+			InMemoryDataStore memoryDataStore = new InMemoryDataStore("testdb");
+			
+			
+
+			//Create a mock StreamFetcher and add it to StreamFetcherManager
+			StreamFetcher streamFetcher = Mockito.mock(StreamFetcher.class);
+			Broadcast stream =  Mockito.mock(Broadcast.class);
+
+			stream.setStreamId(String.valueOf((Math.random() * 100000)));
+
+			stream.setStreamUrl("anyurl");
+			streamFetcher.setStream(stream);
+			when(streamFetcher.getStream()).thenReturn(stream);
+			when(streamFetcher.isStreamAlive()).thenReturn(true);
+			when(streamFetcher.getCameraError()).thenReturn(new Result(true));
+			
+			StreamFetcherFactory factory = mock(StreamFetcherFactory.class);
+			when(factory.make(stream, appScope)).thenReturn(streamFetcher);
+
+			StreamFetcherManager fetcherManager = new StreamFetcherManager(scheduler, memoryDataStore, appScope, factory);
+
+			//set checker interval to 3 seconds
+			fetcherManager.setStreamCheckerInterval(4000);
+
+			//set restart period to 5 seconds
+			fetcherManager.setRestartStreamFetcherPeriod(5);
+
+			//Start stream fetcher
+			Result result = fetcherManager.startStreaming(stream);
+			assertTrue(result.isSuccess());
+
+			//wait 10-12 seconds
+			Thread.sleep(13000);
+
+			//check that stream fetcher stop and start stream is called 4 times
+			verify(streamFetcher, times(2)).stopStream();
+			
+			//it is +1 because it is called at first start
+			verify(streamFetcher, times(3)).startStream(); 
+			verify(streamFetcher, times(3)).startStream(); 
+
+			//set restart period to 0 seconds
+			fetcherManager.setRestartStreamFetcherPeriod(0);
+
+			//wait 10-12 seconds
+			Thread.sleep(13000);
+
+			//check that stream fetcher stop and start stream is not called
+			verify(streamFetcher, times(2)).stopStream();
+			verify(streamFetcher, times(3)).startStream(); 
+			
+			//set restart period to 0 seconds
+			fetcherManager.setRestartStreamFetcherPeriod(5);
+			
+			//wait 10-12 seconds
+			Thread.sleep(13000);
+			
+			//check that stream fetcher stop and start stream is not called
```

Comment:
There should be a separate test for each verifying a feature. Ex: each verify after Thread.sleeps look like testing a different thing.

Reply:
Actually this is a kind of scenario testing.  There is periodic task in StreamFetcher, we change some metrics on the fly and verify periodic task is working properly. Change again some metrics and verify test case again. This is real use case scenario.  We need to sleep sometime to make sure the periodic task do its job.